src/CPUPipe.cpp

```diff
@@ -348,6 +348,24 @@ void CPUPipe::push_input_convolution(unsigned int /*filter_size*/,
     m_batchnorm_stddivs.push_back(variances);
 }
 
+void CPUPipe::move_residual(unsigned int /*filter_size*/,
+                            unsigned int /*channels*/,
+                            unsigned int /*outputs*/,
+                            std::vector<float>&& weights_1,
+                            std::vector<float>&& means_1,
+                            std::vector<float>&& variances_1,
+                            std::vector<float>&& weights_2,
+                            std::vector<float>&& means_2,
+                            std::vector<float>&& variances_2) {
+    m_conv_weights.push_back(std::move(weights_1));
```

Comment:
This is the case where you *really* want emplace_back, isn't it? So you get a move constructor, not a copy constructor...?

Reply:
Also fix the input convolution?

Reply:
emplace_back(std::move(…)) works.
emplace_back(…) still consumes peak memory footprint.

Also fixed convolution input but the gains should be much smaller.